kong/db/schema/typedefs.lua

```diff
@@ -539,12 +538,12 @@ local function validate_path_with_regexes(path)
 end
 
 
-local function validate_regex_or_plain_pattern(pattern)
-  if not is_regex_pattern(pattern) then
+local function validate_header_regex_or_plain_pattern(pattern)
+  if pattern:sub(1, 2) ~= "~*" then
```

Comment:
How about defining a function for the pattern match or add comments for making it clearer?

Reply:
@randmonkey Not likely we are going to have a second use of the function